core/templates/dev/head/pages/signup/Signup.js

```diff
@@ -173,9 +173,48 @@ oppia.controller('Signup', [
       $http.post(_SIGNUP_DATA_URL, requestParams).then(function() {
         window.location = window.decodeURIComponent(
           UrlService.getUrlParams().return_url);
-      }, function() {
+      }, function(rejection) {
+        var loggedOutInNewTabErrorMessage = (
+          'Sorry, you have been logged out [probably in another ' +
+          'window]. Please log in again. You will be redirected ' +
+          'to main page in a while!');
+        if (
+          rejection.data && (
+            rejection.data.error === loggedOutInNewTabErrorMessage)) {
+          $scope.showRedirectToLoginModal();
+        }
         $scope.submissionInProcess = false;
       });
     };
+
+    $scope.showRedirectToLoginModal = function() {
```

Comment:
This modal should not be closable by the user (either by the Esc key or by clicking outside it).

Reply:
Done.